Sources/NIOContext/EventLoopContextCarrier.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  File.swift
+//  
+//
+//  Created by Nathan Harris on 9/16/20.
+//
+
+import Context
+import NIO
+
+public protocol EventLoopContextCarrier: Context {
+    var eventLoop: EventLoop { get }
```

Comment:
How could you make use of this _safely_ when given a context that does not conform to `EventLoopContextCarrier` already?

Following scenario:

- I'm in a CLI app and the context just has a logger.

```swift
import Context
import Logging

let logger = Logger("awesome-cli")
let context = DefaultContext.background(logger: logger)

simple(context) // just needs a logger
```

- Now I'm adding this one library that I need to call, it has this context that requires `EventLoopContextCarrier`

```
import SomeLib 
// someLib(context: EventLoopContextCarrier)
```

- Okey so I guess I have to use `SomeLib.Context` -- seems it conforms to that... okey, let's do that

```diff
- let context = DefaultContext.background(logger: logger)
+ let context = SomeLibContext.background(logger: logger, eventLoop: eventLoop)

simple(context) // just needs a logger
someLib(context: EventLoopContextCarrier)
```

Now I also want to use `OtherLib`, it has some other weird requirement:

```diff
+import OtherLib

let context = SomeLibContext.background(logger: logger, eventLoop: eventLoop)
+// OtherLib.Context: OtherRequirement
+// protocol OtherRequirement { var other: Other {get} }
+let otherContext = ... 
// uhh, that's painful I can't "just conform SomeLibContext to OtherContext's requirement"
// because I'd need to pass in something to it...

simple(context) // just needs a logger
someLib(context) // EventLoopContextCarrier
```

I could pass it into the `context.baggage[OtherKey.self]` and then conform the `SomeLibContext` to that new requirement using this very nasty implementation:

```
extension SomeLibContext: OtherRequirement { 
  var other: Other { self[OtherKey.self]! } // that ! sucks big time, I lost compile time safety!
} 
```

That's pretty bad though... 

- really, we just lost all compile time safety. I claim this will work, but it MAY NOT, because who knows if the baggage/context contains this value actually. 
- It also sucks that the other lib MUST make it's key types public; I believe that's an anti pattern; public API is public API, if it requires things it should require it properly in function calls, not in this weird hidden way.

So context becomes a pain in the neck, for each library which does these "carrier tricks" it becomes more and more complex. We MUST avoid this at any cost. 

Context passing must be boring and simple, a lot of concurrency and tracing depends on it being trivial to do, if it's hard and involves type hackery people won't like it and will just ignore it or get it slightly wrong.